src/common/otel/metrics/taskMetrics.ts

```diff
@@ -0,0 +1,166 @@
+import { IConfig } from 'config';
+import { SERVICES } from '../../constants';
+
+export interface TaskMetricLabels {
+  jobType: string;
+  taskType: string;
+  status?: OperationStatus;
+  errorType?: string;
+}
+
+@singleton()
+export class TaskMetrics {
+  // Task processing metrics
+  private tasksProcessedCounter?: Counter;
+  private tasksProcessingDuration?: Histogram;
+  private tasksSuccessCounter?: Counter;
+  private tasksFailureCounter?: Counter;
+
+  // Gauge metrics
+  private activeTasksGauge?: Gauge;
+
+  // Configuration
+  private readonly taskBuckets: number[];
+  private readonly metricsEnabled: boolean;
+
+  public constructor(
+    @inject(SERVICES.CONFIG) private readonly config: IConfig,
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.METRICS_REGISTRY) private readonly registry?: Registry
+  ) {
+    this.metricsEnabled = this.config.get<boolean>('telemetry.metrics.enabled');
+    this.taskBuckets = this.config.get<number[]>('telemetry.metrics.buckets');
+
+    if (this.registry && this.metricsEnabled) {
+      this.initializeMetrics();
+      this.logger.info('Task metrics initialized');
+    }
+  }
+
+  // === Task Processing ===
+  public recordTaskProcessing(duration: number, labels: TaskMetricLabels): void {
+    if (!this.metricsEnabled) {
+      return;
+    }
+
+    try {
+      const { jobType, taskType, status = OperationStatus.COMPLETED } = labels;
+
+      // Record duration
+      this.tasksProcessingDuration?.observe({ jobType, taskType, status }, duration);
+
+      // Increment processed counter
+      this.tasksProcessedCounter?.inc({ jobType, taskType });
+
+      // Increment status-specific counters
+      if (status === OperationStatus.COMPLETED) {
+        this.tasksSuccessCounter?.inc({ jobType, taskType });
+      } else if (status === OperationStatus.FAILED) {
+        this.tasksFailureCounter?.inc({ jobType, taskType, errorType: labels.errorType ?? 'UnknownError' });
+      }
+    } catch (error) {
+      this.logger.error('Failed to record task processing metrics', { error, labels });
+    }
+  }
+
+  // === Active Tasks Management ===
+  public incrementActiveTasks(labels: Pick<TaskMetricLabels, 'jobType' | 'taskType'>): void {
+    if (!this.metricsEnabled) {
+      return;
+    }
+
+    try {
+      this.activeTasksGauge?.inc(labels);
+    } catch (error) {
+      this.logger.error('Failed to increment active tasks gauge', { error, labels });
+    }
+  }
+
+  public decrementActiveTasks(labels: Pick<TaskMetricLabels, 'jobType' | 'taskType'>): void {
+    if (!this.metricsEnabled) {
+      return;
+    }
+
+    try {
+      this.activeTasksGauge?.dec(labels);
+    } catch (error) {
+      this.logger.error('Failed to decrement active tasks gauge', { error, labels });
+    }
+  }
+
+  // === Utility Methods ===
+  public async withTaskMetrics<T>(labels: Pick<TaskMetricLabels, 'jobType' | 'taskType'>, fn: () => Promise<T>): Promise<T> {
```

Comment:
this is the only public function of the class that is used in other code
consider classifying other functions as private